opentelemetry-rails/lib/opentelemetry/rails/railtie.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module Rails
+    # Railtie automatically configure the OpentTelemetry SDK
+    #
+    # This railtie will set defaults for the following environment variables:
+    # `OTEL_SERVICE_NAME` - if unset, will default to the Rails application name
+    # `OTEL_RESOURCE_ATTRIBUTES` - if unset, will set the deployment environment to the current Rails environment
+    #
+    # This will also set the `OpenTelemetry.logger` to the Rails logger.
+    class Railtie < ::Rails::Railtie
+      railtie_name :opentelemetry
+      config.opentelemetry = ActiveSupport::OrderedOptions.new
+      initializer 'opentelemetry.configure' do |app|
```

Comment:
(apols, link to NR code https://github.com/newrelic/newrelic-ruby-agent/blob/6db6d3fef6c3a493a962794e4d5acd5e2da8f016/lib/newrelic_rpm.rb#L23)

Reply:
hmmm... I need to dig into this a bit more to understand what effect this will have. 

It seems New Relic is dealing with an issue where their instrumentation is inadvertently loading the framework but I have not looked closely enough to see why they are running into this problem:

https://github.com/newrelic/newrelic-ruby-agent/issues/662